src/haddock/modules/analysis/seletopclusts/__init__.py

```diff
@@ -34,20 +30,32 @@ def _run(self):
 
         # how many models should we output?
         models = []
-        for target_ranking in self.params['top_cluster']:
-            if self.params['top_models'] == 'all':
+        if not self.params["top_cluster"]:
```

Comment:
What happen if the user inputs `top_cluster = 0` ?

Reply:
good question! let me address that

Reply:
Added an extra check, thanks!